kernel/exports/library.c

```diff
@@ -281,20 +290,24 @@ klibrary_t * library_open(const char * name, const char * fn) {
     return lib;
 }
 
-int library_close(klibrary_t * lib) {
+int library_close(klibrary_t *lib) {
     // Make sure it's valid.
     if(lib == NULL || lib->lib_close == NULL) {
         errno = EINVAL;
         return -1;
     }
 
+    // Check for reference
+    if(--lib->refcnt > 0) {
```

Comment:
That's the thing, most often you only have to do something when loading and unloading a library, not every time it's opened and closed. I came to this conclusion through experience. Perhaps it would make sense if these were kernel modules, but these are no longer kernel modules, read below.
Same goes for `refcnt`, there is no reason to do it inside the library, none at all. It only brings confusion to the overall system. I'll fix the docs.

Reply:
Ok, excellent. It also makes sense that *if* some sort of per-instance work was needed it could still be done without that (and this way simplifies things).